auctioneer-template/program/src/bid/mod.rs

```diff
@@ -0,0 +1,174 @@
+
+    /// CHECK: Not dangerous. Account seeds checked in constraint.
+    /// Buyer escrow payment account PDA.
+    #[account(
+        mut,
+        seeds = [
+            PREFIX.as_bytes(),
+            auction_house.key().as_ref(),
+            wallet.key().as_ref()
+        ], seeds::program=auction_house_program,
+        bump = escrow_payment_bump
+    )]
+    escrow_payment_account: UncheckedAccount<'info>,
+
+    /// CHECK: Verified with has_one constraint on auction house account.
+    /// Auction House instance authority account.
+    authority: UncheckedAccount<'info>,
+
+    /// Auction House instance PDA account.
+    #[account(seeds = [PREFIX.as_bytes(), auction_house.creator.as_ref(), auction_house.treasury_mint.as_ref()], seeds::program=auction_house_program, bump = auction_house.bump, has_one = authority, has_one = treasury_mint, has_one = auction_house_fee_account)]
+    auction_house: Box<Account<'info, AuctionHouse>>,
+
+    /// CHECK: Not dangerous. Account seeds checked in constraint.
+    /// Auction House instance fee account.
+    #[account(mut, seeds = [PREFIX.as_bytes(), auction_house.key().as_ref(), FEE_PAYER.as_bytes()], seeds::program=auction_house_program, bump = auction_house.fee_payer_bump)]
+    auction_house_fee_account: UncheckedAccount<'info>,
+
+    /// CHECK: Not dangerous. Account seeds checked in constraint.
+    /// Buyer trade state PDA.
+    #[account(mut, seeds = [PREFIX.as_bytes(), wallet.key().as_ref(), auction_house.key().as_ref(), token_account.key().as_ref(), treasury_mint.key().as_ref(), token_account.mint.as_ref(), buyer_price.to_le_bytes().as_ref(), token_size.to_le_bytes().as_ref()], seeds::program=auction_house_program, bump = trade_state_bump)]
+    buyer_trade_state: UncheckedAccount<'info>,
+
+    /// CHECK: Is used as a seed for ah_auctioneer_pda.
+    /// The auctioneer program PDA running this auction.
+    pub auctioneer_authority: UncheckedAccount<'info>,
+
+    /// CHECK: Not dangerous. Account seeds checked in constraint.
+    /// The auctioneer PDA owned by Auction House storing scopes.
+    #[account(
+        seeds = [
+            AUCTIONEER.as_bytes(),
+            auction_house.key().as_ref(),
+            auctioneer_authority.key().as_ref()
+        ], seeds::program=auction_house_program,
+        bump = auction_house.auctioneer_pda_bump,
+    )]
+    pub ah_auctioneer_pda: UncheckedAccount<'info>,
+
+    token_program: Program<'info, Token>,
+    system_program: Program<'info, System>,
+    rent: Sysvar<'info, Rent>,
+}
+
+// Create a private bid on a specific SPL token that is *held by a specific wallet*.
+// pub fn auctioneer_buy<'info>(
+//     ctx: Context<'_, '_, '_, 'info, AuctioneerBuy<'info>>,
+//     trade_state_bump: u8,
+//     escrow_payment_bump: u8,
+//     auctioneer_authority_bump: u8,
+//     buyer_price: u64,
+//     token_size: u64,
+// ) -> Result<()> {
+//     assert_auction_active(&ctx.accounts.listing_config)?;
```

Comment:
So really just this block of 4 function calls and 2 assignments to ctx.accounts.listing_config can be removed (L127 to L133).